cinder/autogenerated_client.go

```diff
@@ -0,0 +1,1821 @@
+package cinder
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type RequestHandlerFn func(*http.Request) (*http.Response, error)
+
+type UpdateVolumeTypeParams struct {
+
+	// VolumeType is required.
+	//
+	// A volume type offers a way to categorize or group volumes.
+	VolumeType string `json:"volume_type"`
+
+	//
+	// The unique identifier of the tenant or account.
+	TenantId string `json:"-"`
+
+	// VolumeTypeId is required.
+	//
+	// The unique identifier for an existing volume type.
+	VolumeTypeId string `json:"-"`
+}
+
+type UpdateVolumeTypeResults struct {
+	VolumeType struct {
+		ExtraSpecs struct {
+			Capabilities string `json:"capabilities"`
+		} `json:"extra_specs"`
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	} `json:"volume_type"`
+}
+
+//
+// Updates a volume type.
+func updateVolumeType(request RequestHandlerFn, args UpdateVolumeTypeParams) (*UpdateVolumeTypeResults, error) {
```

Comment:
@katco-: If you do go ahead and export them, I think it'd be nice to have the generated and hand-written code in separate packages. Along the lines of the "gen" subpackage in https://github.com/awslabs/aws-sdk-go. Doing that would make it very clear that the functions are auto-generated, as opposed to part of a well-tested, curated set of functions.